pydatastructs/strings/string_matching_algorithms.py

```diff
@@ -0,0 +1,136 @@
+from pydatastructs.linear_data_structures.arrays import (
+    OneDimensionalArray)
+
+__all__ = [
+    'find_string'
+]
+
+def find_string(text: str, pattern: str, algorithm: str) -> bool:
+    """API of finding occurrence of a pattern string within another string or body of text.
+
+    Parameters
+    ----------
+    text: str
+        A text, set of characters can include alphabets, numbers , special characters and blank spaces
+    pattern: str
+        A text, set of characters can include alphabets, numbers , special characters and blank spaces
+    algorithm: str
+        A valid algorithm name
+
+    Returns
+    -------
+    bool
+        True if pattern occurs in the string, else False
+
+    Examples
+    --------
+    >>> from pydatastructs.strings.string_matching_algorithms import find_string
+    >>> find_string("aefoaefcdaefcdaed", "aefcdaed", algorithm = "kmp")
+    True
+    >>> find_string("aefoaefcdaefcdaed", "aefcdaedz", algorithm = "kmp")
+    False
+
+    """
+    return eval(algorithm + "('" + text + "','" + pattern + "')")
```

Comment:
Avoid using eval. Please use the pattern similar to the one shown below,

https://github.com/codezonediitj/pydatastructs/blob/0dd2c03419f38e12983f6ec6e4220e4007c811a0/pydatastructs/graphs/algorithms.py#L314-L321